woocommerce/woocommerce
# Marketplace - tracking enhancements

### Changes proposed in this Pull Request:





Closes https://github.com/Automattic/woocommerce.com/issues/18619

This PR adds some enhanced tracking to some marketplace interactions. These events all respect the WooCommerce option to allow/disallow tracking and collect no personally identifying information.

Added the following interactions:
- Event tracking what products were displayed on the "Discover" tab
- Clicks on product cards with useful data
- View all buttons on search page clicked
- Category selector "more" dropdowns clicked



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build assets and [visit the new marketplace page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions)
2. Go to **WooCommerce > Settings > Advanced > WooCommerce.com** and make sure **Allow usage of WooCommerce to be tracked** option is checked
3. Open JS console and run `localStorage.setItem( 'debug', 'wc-admin:*' );`. This makes tracks logs show up in console. While in development, tracks events are not sent up to the server.
4. On the dev console, click on **Preserve Logs**. So that even if the page reload, the recorded console logs are preserved.
5.  Run these actions:
    - Load the [discover page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions). 
        - Event name is `wcadmin_marketplace_discover_viewed`
        - This will send all shown products ids in the event
    - Load the [browse page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=extensions&path=%2Fextensions) and click more on the category selector
        - Event name is `marketplace_category_dropdown_opened`
        - The only property is `type` which can be `extensions` or `themes`
    -  On the product page, click on the vendor name in the product cards. This will take you WooCommerce.com
        - Event name is `wcadmin_marketplace_product_card_vendor_clicked`
        - Since we're actually closing this page to go to a new one, the queue system will take place and in the console it'll say **Processing items in queue** and it'll go to WooCommerce.com. 
        - From WooCommerce.com, go back to the store admin. The console will say **Processing item in queue**. You can look at the arguments for this and see the event name.
      - On the product page, click on the product card. This will take you WooCommerce.com
          - Event name is: `"marketplace_product_card_clicked"`
          - Please repeat the steps we did with the vendor name click
          - If you click on a product card after performing a text search, there should be `search_term` property in Tracks data
          - If you click on a product card after selecting a category, there should be `category` property in Tracks data
6. Please test around the issue

### Screenshot
![image](https://github.com/woocommerce/woocommerce/assets/10389957/9d413bfe-f3ab-4c55-9e02-a6b53ffe9e98)
